internal/validation/apiextensions/v1/xrd/handler_test.go

```diff
@@ -0,0 +1,346 @@
+/*
+Copyright 2023 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package xrd
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/fieldpath"
+	"github.com/crossplane/crossplane-runtime/pkg/test"
+
+	v1 "github.com/crossplane/crossplane/apis/apiextensions/v1"
+)
+
+var _ admission.CustomValidator = &validator{}
+
+func TestValidateUpdate(t *testing.T) {
```

Comment:
I didn't add them because the handler returns a single error and all the information contained in the `field.ErrorList` is squashed to a single `error`, which is hard to inspect. Negative tests are better handled as part of `TestValidateUpdate` on the XRD's method: https://github.com/crossplane/crossplane/pull/4484/files#diff-73fd2b803fc7540f5fadc2b9eb0f0c9d5f5b10d02b5e3a88e7ac1c70618d8016L33.
However I agree I should add more failures related to the dry-run, replicating the ones for the creation.

Reply:
added a few tests 👍